internal/store/custom_dns.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+//go:generate mockgen -destination=../mocks/mock_custom_dns.go -package=mocks github.com/mongodb/mongocli/internal/store CustomDNSUpdater,CustomDNSDescriber
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongocli/internal/config"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+type CustomDNSUpdater interface {
+	UpdateCustomDNS(string, *atlas.AWSCustomDNSSetting) (*atlas.AWSCustomDNSSetting, error)
+}
+
+type CustomDNSDescriber interface {
+	DescribeCustomDNS(string) (*atlas.AWSCustomDNSSetting, error)
+}
+
+// UpdateCustomDns encapsulates the logic to manage different cloud providers
+func (s *Store) UpdateCustomDNS(projectID string, customDNSSetting *atlas.AWSCustomDNSSetting) (*atlas.AWSCustomDNSSetting, error) {
```

Comment:
[q] could we instead of having an `CustomDNSUpdater` have a `CustomDNSEnabler` and a `CustomDNSDisabler` given and manage the logic in here (the store)? this would simplify what the command does as they don't need much more than to call the right store